tests/vic-machine.bats

```diff
@@ -12,135 +13,126 @@
+    run "$installer" -target "$GOVC_URL" -user root
     assert_failure
-    echo ${lines[0]}
-    [[ ${lines[0]} =~ "-compute-resource Compute resource path must be specified" ]]
+    assert_line -e "-compute-resource Compute resource path must be specified"
 }
 
 @test "vic-machine image-store is missing" {
-    run "${installer}" -target ${TEST_URL} -user root -compute-resource $POOL_PATH
+    run "$installer" -target "$GOVC_URL" -user root -compute-resource "$GOVC_RESOURCE_POOL"
     assert_failure
-    echo ${lines[0]}
-    [[ ${lines[0]} =~ "-image-store Image datastore name must be specified" ]]
+    assert_line -e "-image-store Image datastore name must be specified"
 }
 
 @test "vic-machine deployment and docker commands" {
-    params="-target $TEST_URL -name ${vch_name} -user ${USER} -compute-resource $POOL_PATH -image-store $IMAGE_STORE_NAME"
-    if [ $ISVC == 1 ]; then
-	  assert [ -n $BRIDGE_NETWORK ]
-	  params="$params -bridge-network $BRIDGE_NETWORK"
-	fi
-    echo "${installer}" $params
-    params="$params -passwd ${PASSWORD}"
-    run "${installer}" $params
-    assert_success
-    len=${#lines[@]}
-    echo ${lines[len-2]}
-    echo ${lines[len-1]}
-    [[ ${lines[len-1]} =~ "Installer completed successfully..." ]]
-    docker_cmd=`echo ${lines[len-2]} | awk '{$1=$2=""; $(NF--)=""; print$0}'`
-    echo "docker_cmd="$docker_cmd
-    # execute sample docker command
-    run "$docker" $docker_cmd info
+    params=(-name $vch_name -target $GOVC_URL -user $GOVC_USERNAME -passwd $GOVC_PASSWORD
+            -compute-resource $GOVC_RESOURCE_POOL -image-store $GOVC_DATASTORE)
+
+    if [ "$(govc about -json | jq -r .About.ApiType)" = "VirtualCenter" ] ; then
+        assert [ -n "$BRIDGE_NETWORK" ]
+        params+=(-bridge-network $BRIDGE_NETWORK)
+    fi
+
+    local output status
+    run "$installer" "${params[@]}"
     assert_success
-    echo ${lines[0]}
-
-    #reinstall with same name
-    params="-target $TEST_URL -name ${vch_name} -user ${USER} -compute-resource $POOL_PATH -image-store $IMAGE_STORE_NAME -force -key ./${vch_name}-key.pem -cert ./${vch_name}-cert.pem"
-    if [ $ISVC == 1 ]; then
-	  params="$params -bridge-network $BRIDGE_NETWORK"
-	fi
-    echo "${installer}" $params
-    params="$params -passwd ${PASSWORD}"
-    run "${installer}" $params
+    assert_line -e "Installer completed successfully"
+    docker_cmd=($(grep 'docker -H' <<<"$output" | cut -d' ' -f3-7))
+
+    # execute docker info command
+    for _ in $(seq 1 5) ; do
+      run "$docker" "${docker_cmd[@]}" info
+
+      if [ "$status" -eq 0 ] ; then
+          assert_line "Name: VIC"
+          break
+      fi
+
+      # retry as listener may not be up yet
+      sleep 1
+    done
+
     assert_success
-    len=${#lines[@]}
-    echo ${lines[len-1]}
-    [[ ${lines[len-1]} =~ "Installer completed successfully..." ]]
-    docker_cmd=`echo ${lines[len-2]} | awk '{$1=$2=""; $(NF--)=""; print$0}'`
+
```

Comment:
Looks like you removed the second installation in this test case. I intentionally rerun the installation, to make sure -force and -generate-cert=false work here.